cura/Settings/MachineManager.py

```diff
@@ -489,22 +395,70 @@ def activeMaterialId(self):
 
         return ""
 
+    @pyqtProperty("QVariantMap", notify = activeMaterialChanged)
+    def allActiveMaterialIds(self):
+        if not self._global_container_stack:
+            return {}
+
+        result = {}
+
+        stacks = [ s for s in ExtruderManager.getInstance().getMachineExtruders(self._global_container_stack.getId()) ]
+        stacks.insert(0, self._global_container_stack)
+
+        for stack in stacks:
+            material_container = stack.findContainer(type = "material")
+            if not material_container:
+                continue
+
+            key = ""
+            if stack == self._global_container_stack:
+                key = "global"
```

Comment:
Why not `self._global_container_stack.getId()`? No need for this extra case, right?